mavros/src/plugins/param.cpp

```diff
@@ -48,179 +54,135 @@ using utils::enum_value;
-  using XmlRpcValue = XmlRpc::XmlRpcValue;
 
-  std::string param_id;
-  XmlRpcValue param_value;
-  uint16_t param_index;
   uint16_t param_count;
+  uint16_t param_index;
+
+  Parameter() : rclcpp::Parameter() {}
 
-  void set_value(mavlink::common::msg::PARAM_VALUE & pmsg)
+  template<typename ValueTypeT>
+  Parameter(std::string const& name, ValueTypeT value, int count, int index)
+    : rclcpp::Parameter(name,value),
+    param_count(count),
+    param_index(index)
   {
+  }
+
+  static Parameter from_mavlink(mavlink::common::msg::PARAM_VALUE & pmsg) {
     mavlink::mavlink_param_union_t uv;
     uv.param_float = pmsg.param_value;
 
-    // #170 - copy union value to itermediate var
-    int int_tmp;
-    float float_tmp;
+    auto param_name = mavlink::to_string(pmsg.param_id);
 
     switch (pmsg.param_type) {
       // [[[cog:
       // param_types = [ (s, 'float' if s == 'real32' else s) for s in (
-      //     'int8', 'uint8',
-      //     'int16', 'uint16',
-      //     'int32', 'uint32',
+      //     'int8',
+      //     'int16',
+      //     'int32',
       //     'real32',
       // )]
-      // unsupported_types = ('int64', 'uint64', 'real64')
+      // unsupported_types = ('uint8', 'uint16', 'uint32', 'int64', 'uint64', 'real64')
       //
       // for a, b in param_types:
-      //     btype = 'int' if 'int' in b else b
       //     cog.outl("case enum_value(MT::%s):" % a.upper())
-      //     cog.outl("\t%s_tmp = uv.param_%s;" % (btype, b))
-      //     cog.outl("\tparam_value = %s_tmp;" % btype)
-      //     cog.outl("\tbreak;")
+      //     cog.outl("\treturn Parameter(param_name, uv.param_%s, pmsg.param_count, pmsg.param_index);" % (b))
       // ]]]
       case enum_value(MT::INT8):
-        int_tmp = uv.param_int8;
-        param_value = int_tmp;
-        break;
-      case enum_value(MT::UINT8):
-        int_tmp = uv.param_uint8;
-        param_value = int_tmp;
-        break;
+      	return Parameter(param_name, uv.param_int8, pmsg.param_count, pmsg.param_index);
       case enum_value(MT::INT16):
-        int_tmp = uv.param_int16;
-        param_value = int_tmp;
-        break;
-      case enum_value(MT::UINT16):
-        int_tmp = uv.param_uint16;
-        param_value = int_tmp;
-        break;
+      	return Parameter(param_name, uv.param_int16, pmsg.param_count, pmsg.param_index);
       case enum_value(MT::INT32):
-        int_tmp = uv.param_int32;
-        param_value = int_tmp;
-        break;
-      case enum_value(MT::UINT32):
-        int_tmp = uv.param_uint32;
-        param_value = int_tmp;
-        break;
+      	return Parameter(param_name, uv.param_int32, pmsg.param_count, pmsg.param_index);
       case enum_value(MT::REAL32):
-        float_tmp = uv.param_float;
-        param_value = float_tmp;
-        break;
-      // [[[end]]] (checksum: 5950e4ee032d4aa198b953f56909e129)
+      	return Parameter(param_name, uv.param_float, pmsg.param_count, pmsg.param_index);
```

Comment:
That part of conversions should work - ints are convertible to PARAMETER_INTEGER.